File: app/src/main/java/com/jamieadkins/gwent/deck/detail/user/DeckBuilderContract.java
package com.jamieadkins.gwent.deck.detail.user;

import com.jamieadkins.gwent.card.list.CardsContract;
import com.jamieadkins.gwent.deck.detail.DeckDetailsContract;

/**
 * Specifies the contract between the view and the presenter.
 */

public interface DeckBuilderContract {
    interface View extends DeckDetailsContract.View, CardsContract.View {

    }

    interface Presenter extends DeckDetailsContract.Presenter {

    }
}

File: app/src/main/java/com/jamieadkins/gwent/deck/detail/DeckBuilderContract.kt
package com.jamieadkins.gwent.deck.detail

import com.jamieadkins.gwent.card.list.CardsContract

/**
 * Specifies the contract between the view and the presenter.
 */

interface DeckBuilderContract {
    interface CardDatabaseView : View, CardsContract.View

    interface View

    interface Presenter : CardsContract.Presenter
}
